Inits new doc pages for existing Vincent Abilities and Policies (#268)
LIT-Protocol/Vincent: docs/src/Developers/Abilities/Morpho.md

---
title: Morpho
---

# Morpho

> The Morpho Ability enables Vincent Apps to interact with Morpho lending vaults on behalf of Vincent Users using their Vincent Agent Wallets. This allows Vincent Apps to provide yield-generating DeFi functionality through Morpho's lending markets, including depositing assets to earn yield, withdrawing funds, and managing vault positions.

Reviewer: Same suggestion as above
```suggestion
The Morpho Ability enables Vincent Apps to interact with Morpho lending vaults on behalf of Vincent Users. This allows Vincent Apps to provide yield-generating DeFi functionality through Morpho's lending markets, including depositing assets to earn yield, withdrawing funds, and managing vault positions.
```

## Key Features

- **Yield Vault Operations**: Supports deposit, withdraw, and redeem operations with Morpho's ERC-4626 compliant vaults using Vincent Agent Wallets within a secure Trusted Execution Environment
- **Multi-Chain Support**: Works across multiple networks where Morpho vaults are deployed including Ethereum, Base, Arbitrum, Optimism, and Polygon
- **Comprehensive Vault Discovery**: Utilizes the Morpho Vault API to discover and filter vaults by asset, chain, APY, TVL, and other criteria
- **Gas Sponsorship**: Optional integration with Alchemy's gas sponsorship for gasless transactions via EIP-7702

## How It Works

The Morpho Ability is built using the [Vincent Ability SDK](../Ability-Developers/Creating-Abilities.md) and operates in two phases:

1. **Precheck Phase**: Validates all prerequisites for the Morpho operation

   - Validates operation type, vault address, and amount format
   - Verifies the vault exists using the built-in Morpho Vault client and retrieves asset information

>    - For deposits, checks the Vincent App User's Agent Wallet's token balance and allowance

Reviewer: Maybe simplify to

```suggestion
   - For deposits, checks the User's Agent Wallet token balance and allowance
```

   - For withdrawals/redeems, checks the Vincent App User's Agent Wallet's has a vault share balance

>    - For withdrawals, converts the requested asset amount to required vault shares using ERC-4626 `convertToShares`

Reviewer: ```suggestion
   - For withdrawals, converts the requested asset amount to the required vault shares using ERC-4626 `convertToShares`
```
